monai/deploy/operators/dicom_data_loader_operator.py

```diff
@@ -171,34 +174,94 @@ def _load_data(self, files: List[str]):
+
+                # Prepare sop_map entry
+                series_key = (study_instance_uid, series_instance_uid)
+                sop_map.setdefault(series_key, {})
+                ipp = sop_instance.get("ImagePositionPatient", None)
+                if ipp is not None:
+                    # Convert IPP to tuple
+                    ipp_tuple = cast(Tuple[float, float, float], tuple(float(v) for v in ipp))
+                else:
+                    # Non-image files will be missing IPP; store SOP instance under "None" key, move on to next SOP instance
+                    sop_map[series_key].setdefault(ipp, []).append(sop_instance)
+                    continue
+
+                sop_list = sop_map[series_key].setdefault(ipp_tuple, [])
+
+                if not sop_list:
+                    # First occurrence of this spatial position — store the SOP instance
+                    sop_list.append(sop_instance)
+                else:
+                    # Duplicate spatial location found — compare AcquisitionNumbers (if absent, set to -1)
+                    exist = sop_list[0]
+                    exist_acq_num = int(exist.get("AcquisitionNumber", -1))
+                    curr_acq_num = int(sop_instance.get("AcquisitionNumber", -1))
+                    if curr_acq_num > exist_acq_num:
+                        # Current SOP instance AcquisitionNumber is greater - replace existing SOP instance
+                        self._logger.info(
+                            f"Duplicate spatial coordinates detected; removing duplicate SOP at IPP {ipp_tuple} "
+                            f"in Series {series_instance_uid}; removed SOP instance with lower AcquisitionNumber "
+                            f"({curr_acq_num} < {exist_acq_num})"
+                        )
+                        sop_list[0] = sop_instance
+                    elif curr_acq_num < exist_acq_num:
+                        # Existing SOP instance AcquisitionNumber is greater - don't store current SOP instance
+                        self._logger.info(
+                            f"Duplicate spatial coordinates detected; removing duplicate SOP at IPP {ipp_tuple} "
+                            f"in Series {series_instance_uid}; kept SOP instance with higher AcquisitionNumber "
+                            f"({exist_acq_num} > {curr_acq_num})"
+                        )
+                    elif curr_acq_num == -1:
+                        # AcquisitionNumber tag is absent for compared SOP instances - don't store current SOP instance
+                        self._logger.info(
+                            f"Duplicate spatial coordinates detected; removing duplicate SOP at IPP {ipp_tuple} "
+                            f"in Series {series_instance_uid}; AcquisitionNumber tags are absent"
+                        )
+                    else:
+                        # AcquisitionNumber tag values are equal for compared SOP instances - don't store current SOP instance
```

Comment:
Wonder if we can compare the acquisition time and take the newer one (e.g. assuming the modality is smart and can re-acquire a new image at the same position when it detects the raw data is not good for reconstruction). Acquisition time is not mandatory and can be absent though.

The above would be the edge case of the edge cases, so the implementation as is should suffice.